biojava-integrationtest/src/test/java/org/biojava/nbio/structure/test/io/cif/CifFileConsumerImplTest.java

```diff
@@ -0,0 +1,357 @@
+                " 1omh, 1zog, 2b68, 1nqb, 1t7k")
+                .parallel()
+                .forEach(pdbId -> {
+                    System.out.println(pdbId);
+                    Structure cif = loadLocalCif(pdbId);
+                    assertNumberFormat(cif);
+                    Structure bcif = loadLocalBcif(pdbId);
+                    assertNumberFormat(bcif);
+                });
+    }
+
+    private Structure loadLocalCif(String pdbId) {
+        try {
+            String middle = pdbId.substring(1, 3);
+            return CifFileConverter.convert(CifReader.readText(new GZIPInputStream(Files.newInputStream(Paths.get("/var/pdb/" + middle + "/" + pdbId + ".cif.gz")))));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private Structure loadLocalBcif(String pdbId) {
+        try {
+            String middle = pdbId.substring(1, 3);
+            return CifFileConverter.convert(CifReader.readBinary(Files.newInputStream(Paths.get("/var/bcif/" + middle + "/" + pdbId + ".bcif"))));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void assertNumberFormat(Structure structure) {
+        PDBHeader header = structure.getPDBHeader();
+        assertNotNull(header);
+        Date relDate = header.getRelDate();
+        assertNotNull(relDate);
+        Date modDate = header.getModDate();
+        assertNotNull(modDate);
+    }
+
+    enum Source {
+        BCIF, // binary cif using ciftools
+        CIF, // mmCIF using ciftools
+        MMCIF, // mmCIF using BioJava impl
+        MMTF // mmtf using BioJava impl
+    }
+
+    /**
+     * Performance diary;
+     *
+     * 05/01/19 - ciftools v0.3.0, parallel, bcif, non-gzipped, 12 worker threads
+     * BCIF: 918 s for 151079 structures, 6073 µs per structure, failed for 0 entries
+     * CIF: 2502 s for 151079 structures, 16562 µs per structure, failed for 0 entries
+     *
+     * 05/02/19 -  ciftools v0.3.0, parallel, bcif, non-gzipped for bcif / gzipped for cif, 12 worker threads
+     * BCIF: 35 s for 5000 structures, 6228 µs per structure, failed for 0 entries
+     * CIF: 132 s for 5000 structures, 23902 µs per structure, failed for 0 entries
+     * MMCIF: 310 s for 5000 structures, 52329 µs per structure, failed for 0 entries
+     * MMTF: 17 s for 5000 fetched structures, 3211 µs per structure, failed for 0 entries
+     *
+     * CIF parsing using David's tokenizer approach takes roughly half the time. Binary parsing using ciftools takes
+     * roughly double the time compared to MMTF and should be even more at a disadvantage when MMTF data is stored
+     * locally. However, MMTF provides almost no metadata.
+     *
+     * @throws IOException propagated
+     */
+    @Test
+    @Ignore("ignore long-running test, do run to track performance")
```

Comment:
This test could be moved to the integrationtest module, where we place longer tests that run for longer and can be skipped during regular development.

Reply:
That tests should already be in the integrationtest module - however, there is also a test class of the same name in the structure module. I've renamed the tests in the integration module (1755a4c), so that there are no duplicate name for test classes.

Reply:
Oh, I see! Sorry for the confusion.